### PR TITLE
tests: posix: common: separate posix spinlock tests into a standalone test

### LINKED ISSUE #80965: tests: posix: common: split spinlock tests into a standalone test
**Is your enhancement proposal related to a problem? Please describe.**
Make the spinlock-related tests in `tests/posix/common` into a standalone test.

**Describe the solution you'd like**
The obvious way to group spinlock-related testsuites is by `POSIX_SPIN_LOCKS ` [Option Group](https://pubs.opengroup.org/onlinepubs/9799919799/xrat/V4_subprofiles.html).

**Describe alternatives you've considered**
The way the testsuite has been designed historically.

**Additional context**
See #79454 for more info

### PR DESCRIPTION
**Pull request by Pancakem**
Fixes #80965 

posix.common contains testsuites that can be separated into smaller groups of tests. This change moves spinlock into a singular testsuite at tests/posix/spinlock app directory.

### CODE REVIEW

File: tests/posix/spinlock/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(posix_spinlock)
```
Comment (cfriedt) [inline tests/posix/spinlock/CMakeLists.txt:5]: To match the option group name
```suggestion
project(posix_spinlocks)
```

File: tests/posix/spinlock/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+FILE(GLOB app_sources src/*.c)
```
Comment (cfriedt) [inline tests/posix/spinlock/CMakeLists.txt:7]: This line can be removed
```suggestion
```

File: tests/posix/spinlock/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+zephyr_include_directories(${ZEPHYR_BASE}/lib/posix)
```
Comment (cfriedt) [inline tests/posix/spinlock/CMakeLists.txt:8]: These two lines can be removed
```suggestion
FILE(GLOB app_sources src/*.c)
zephyr_include_directories(${ZEPHYR_BASE}/lib/posix)
```

File: tests/posix/spinlock/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+target_sources(app PRIVATE ${app_sources})
```
Comment (cfriedt) [inline tests/posix/spinlock/CMakeLists.txt:10]:
```suggestion
target_sources(app PRIVATE src/main.c)
```

File: tests/posix/spinlock/CMakeLists.txt
Comment (cfriedt) [inline tests/posix/spinlock/CMakeLists.txt:1]: please rename the directory from `spinlock` to `spinlocks` to match the option group naming

File: tests/posix/spinlock/testcase.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+common:
+  filter: not CONFIG_NATIVE_LIBC
+  tags:
+    - posix
+    - spinlock
```
Comment (cfriedt) [inline tests/posix/spinlock/testcase.yaml:5]:
```suggestion
    - spinlocks
```

File: tests/posix/spinlock/testcase.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  # 1 tier0 platform per supported architecture
+  platform_key:
+    - arch
+    - simulation
+tests:
+  portability.spinlock.rwlocks:
```
Comment (cfriedt) [inline tests/posix/spinlock/testcase.yaml:11]: Please match the option group name for all testsuites below
```suggestion
  portability.posix.spinlocks:
```

File: tests/posix/spinlock/src/spinlock.c
Comment (cfriedt) [inline tests/posix/spinlock/src/spinlock.c:1]: please rename this file to `main.c` and ensure that the test suite and ZTEST lines match `posix_spinlocks`

**Review by cfriedt — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Pull request merged**